torch/nn/parallel/distributed.py

```diff
@@ -815,8 +815,21 @@ def join(self, divide_by_initial_world_size=True, enable=True):
             if enable and not has_error:
                 all_procs_joined = False
                 is_last_joiner = True
-                # Schedules allreduce to match fwd pass allreduce in non-joined procs
+                i = 0
+                WARN_THRESHOLD = 1000
+                warned = False
                 while not all_procs_joined:
+                    if i > WARN_THRESHOLD and not warned:
+                        my_rank = dist.get_rank(self.process_group)
+                        warnings.warn(
+                            "Detected uneven input skew of greater "
+                            f"than {WARN_THRESHOLD}. This means that rank {my_rank} "
+                            f"has at least {WARN_THRESHOLD} fewer inputs than "
+                            "other currently active ranks. This level of skew could "
+                            "lead to performance degradation during training."
+                        )
+                        warned = True
```

Comment:
Please feel free to ignore. This can also be done using `warnings.simplefilter("once")` IIUC.

Reply:
Thanks! This is better than toggling the boolean.